src/esp/metadata/managers/SensorAttributesManager.cpp

```diff
@@ -0,0 +1,259 @@
+}  // SensorAttributesManager::createAttributesFromSensorSpec
+
+AbstractSensorAttributes::ptr SensorAttributesManager::createObject(
+    const std::string& sensorFileName,
+    bool registerTemplate) {
+  auto sensorAttrs = this->createDefaultObject(sensorFileName, false);
+  if (nullptr == sensorAttrs) {
+    return sensorAttrs;
+  }
+  ESP_DEBUG(Mn::Debug::Flag::NoSpace)
+      << "Sensor attributes (" << sensorFileName << ":"
+      << sensorAttrs->getHandle() << ") created"
+      << (registerTemplate ? " and registered." : ".");
+
+  // TODO : set handle appropriately to be registration key
+
+  return this->postCreateRegister(std::move(sensorAttrs), registerTemplate);
+}  // SensorAttributesManager::createObject
+
+AbstractSensorAttributes::ptr SensorAttributesManager::buildObjectFromJSONDoc(
+    const std::string& filename,
+    const io::JsonGenericValue& jsonConfig) {
+  // Get sensor class type from jsonconfig, use this to determine class
+  // name to build
+  std::string tmpStrVal = "";
+  // Look for sensor_subtype string value in json config
+  if (!io::readMember<std::string>(jsonConfig, "sensor_subtype", tmpStrVal)) {
+    // Value not found in json so this is an error - we don't know what kind of
+    // sensor to create
+    return nullptr;
+  }
+  std::string strToLookFor = Cr::Utility::String::lowercase(tmpStrVal);
+  if (strToLookFor == "unspecified") {
+    // cannot instantiate unspecified sensor; error
+    ESP_ERROR(Mn::Debug::Flag::NoSpace)
+        << "Unable to determine Sensor Sub Type in JSON Config `" << filename
+        << "` so unable to build correct SensorAttributes. "
+           "Aborting.";
+    return nullptr;
+  }
+  auto found = attributes::SensorSubTypeNamesMap.find(strToLookFor);
+  if (found == attributes::SensorSubTypeNamesMap.end()) {
+    // string representation of enum value was not found in mappings; error
+    return nullptr;
+  }
+  // By here, legal sensor subtype is found
+  sensor::SensorSubType sensorSubType = found->second;
+
+  const std::string& sensorAttrClassName =
+      SensorAttributesManager::SenssorAttrsTypeNamesMap.at(sensorSubType);
+  CORRADE_ASSERT(!sensorAttrClassName.empty(),
+                 "Unknown SensorAttributes class for SensorSubType enum value :"
+                     << static_cast<int32_t>(sensorSubType),
+                 nullptr);
+
+  // Build and initialize appropriate class of Attributes based on class name
+  auto sensorAttributes =
+      this->initNewObjectInternal(sensorAttrClassName, true);
+  if (nullptr == sensorAttributes) {
+    return sensorAttributes;
+  }
+  // This should never fail
+  this->setValsFromJSONDoc(sensorAttributes, jsonConfig);
+  return sensorAttributes;
+}  // SensorAttributesManager::buildObjectFromJSONDoc
+
+void SensorAttributesManager::setValsFromJSONDoc(
+    AbstractSensorAttributes::ptr attribs,
+    const io::JsonGenericValue& jsonConfig) {
+  // TODO support loading values from JSON docs for each type of
```

Comment:
Would be good to have 1 default for each sensor type, maybe in test_assets? Or programmatically created and added like prims.

Reply:
Yeah, I planned on doing that in the 2nd PR.